### PR TITLE
unit tests for GetDeployer

### PR DESCRIPTION
**Pull request by hmlanigan**
## Description of change

Unit tests for GetDeployer.  Part of an effort to move juju deploy command unit tests to be more focused on individual pieces and not full stack.  Continuing in #11901

### CODE REVIEW

File: cmd/juju/application/deployer/mocks/store_mock.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
+// Code generated by MockGen. DO NOT EDIT.
+// Source: github.com/juju/juju/cmd/juju/application/store (interfaces: CharmrepoForDeploy,MacaroonGetter)
+
+// Package mocks is a generated GoMock package.
+package mocks
+
+import (
+	gomock "github.com/golang/mock/gomock"
+	charm "github.com/juju/charm/v7"
```
Comment (hpidcock) [inline cmd/juju/application/deployer/mocks/store_mock.go:9]: should be for charm/v8

File: go.mod [hunk 2]
```diff
@@ -42,7 +42,9 @@ require (
 	github.com/joyent/gosign v0.0.0-20140524000734-0da0d5f13420
 	github.com/juju/ansiterm v0.0.0-20180109212912-720a0952cc2a
 	github.com/juju/bundlechanges/v2 v2.0.0-20200818094029-ae6d813368b2
+	github.com/juju/charm/v7 v7.0.0-20200424224456-5fe646695e85
```
Comment (hpidcock) [inline go.mod:45]: drop charm/v7

File: go.mod [hunk 3]
```diff
@@ -42,7 +42,9 @@ require (
...
 	github.com/juju/charm/v8 v8.0.0-20200817113526-2a88e9b46b47
+	github.com/juju/charmrepo/v5 v5.0.0-20200424225329-cddcb4fdcd09
```
Comment (hpidcock) [inline go.mod:47]: drop charmrepo/v5

**Review by hpidcock — APPROVED** [review]
Few dependency issues, but LGTM

File: cmd/juju/application/deployer/mocks/store_mock.go (hunk 1 above)
Comment (hmlanigan) [reply cmd/juju/application/deployer/mocks/store_mock.go]: resolved

**Comment by hmlanigan** [comment]
$$merge$$

**Pull request merged**